lib/VolSurface/Calibration/Equities.pm

```diff
@@ -0,0 +1,637 @@
+
+#Do simple expansion or in other words look up a little further in this direction
+            @current_expansion = map { $simplex_centroid[$_] + 2 * ($current_reflection[$_] - $simplex_centroid[$_]) } (0 .. $num_of_var - 1);
+
+            @simplex_centroid =
+                map { $current_reflection[$_] } (0 .. $num_of_var - 1);
+            $expansion_function_eval = $self->function_to_optimize(\@current_expansion);
+            if ($expansion_function_eval < $reflected_function_eval) {
+
+                #replace highest point with expansion point
+                #assing function value to highest point
+                @{$simplex[$highest_point_index]} =
+                    map { $current_expansion[$_] } (0 .. $num_of_var - 1);
+                $function_eval[$highest_point_index] = $expansion_function_eval;
+            } else {
+
+                #replace highest point with reflected point
+                #assign highest value to reflection point
+                @{$simplex[$highest_point_index]} =
+                    map { $simplex_centroid[$_] } (0 .. $num_of_var - 1);
+                $function_eval[$highest_point_index] = $reflected_function_eval;
+            }
+
+        } else {
+            if ($reflected_function_eval < $highest_value) {
+
+                #replace the simplex highest point with reflected point;
+                @{$simplex[$highest_point_index]} =
+                    map { $current_reflection[$_] } (0 .. $num_of_var - 1);
+                $function_eval[$highest_point_index] = $reflected_function_eval;
+            }
+
+            @current_contraction =
+                map { $simplex_centroid[$_] + 0.5 * ($simplex[$highest_point_index][$_] - $simplex_centroid[$_]) } (0 .. $num_of_var - 1);
+            $contraction_function_eval = $self->function_to_optimize(\@current_contraction);
+            if ($contraction_function_eval < $function_eval[$highest_point_index]) {
+                @{$simplex[$highest_point_index]} =
+                    map { $current_contraction[$_] } (0 .. $num_of_var - 1);
+                $function_eval[$highest_point_index] = $contraction_function_eval;
+            } elsif ($reflected_function_eval >= $highest_value) {
+                $calcvert = 1;
+                for ($i = 0; $i <= $num_of_points; $i++) {
+                    if ($i != $lowest_point_index) {
+                        @{$simplex[$i]} =
+                            map { $simplex[$lowest_point_index][$_] + ($simplex[$i][$_] - $simplex[$lowest_point_index][$_]) / 2.0 }
+                            (0 .. $num_of_var - 1);
+                    }
+                    $function_eval[$highest_point_index] = $self->function_to_optimize($simplex[$highest_point_index]);
+                }
+            }
+        }
+        $counter++;
+    }
+
+    my @sorted_eval = sort { $a <=> $b } @function_eval;
+    $lowest_point_index = first_index { $_ == $sorted_eval[0] } @function_eval;
```

Comment:
Even here, the only thing you need is the index of the lowest value. A simple loop scanning the array is probably much faster than the sort plus scan.